core/src/integration-test/java/com/scalar/db/api/DistributedStorageIntegrationTestBase.java

```diff
@@ -1418,6 +1426,142 @@ public void scan_ScanLargeDataWithOrdering_ShouldRetrieveExpectedValues()
     }
   }
 
+  @Test
+  public void scanAll_NoLimitGiven_ShouldRetrieveAllRecords()
+      throws ExecutionException, IOException {
+    // Arrange
+    populateRecords();
+    ScanAll scanAll = new ScanAll();
+
+    // Act
+    List<Result> results = scanAll(scanAll);
+
+    // Assert
+    List<ExpectedResult> expectedResults = new ArrayList<>();
+    IntStream.range(0, 5)
+        .forEach(
+            i ->
+                IntStream.range(0, 3)
+                    .forEach(
+                        j -> {
+                          ExpectedResultBuilder erBuilder =
+                              new ExpectedResultBuilder()
+                                  .partitionKey(Key.ofInt(COL_NAME1, i))
+                                  .clusteringKey(Key.ofInt(COL_NAME4, j))
+                                  .nonKeyColumns(
+                                      ImmutableList.of(
+                                          TextColumn.of(COL_NAME2, Integer.toString(i + j)),
+                                          IntColumn.of(COL_NAME3, i + j),
+                                          BooleanColumn.of(COL_NAME5, j % 2 == 0)));
+                          expectedResults.add(erBuilder.build());
+                        }));
+    assertResultsContainsExactlyInAnyOrder(results, expectedResults);
```

Comment:
Since the `scanAll` results are unordered, asserting the values are correct is not as straightforward as for `Scan` results, so I added two utility methods in `TestUtils` to do so : 
- `assertResultsContainsExactlyInAnyOrder(results, expectedResults)`
- `assertResultsAreASubsetOf(results, expectedResults)`